### ISSUE TITLE
Two questions about regularization 

### ISSUE THREAD

**Issue by mingyr** [deepmind/sonnet#126]
Dear Deepminder:

I have two questions regarding regularization, and thanks a lot for providing your insight into these.

The first is, when I check the code snippet demonstrating how to use regularization, I noticed the bias is also applied for regularization. So my first question is, is it appropriate or not. I remembered it is not recommended to regularize the bias, since it represents some level threshold, like the DC in FFT. Or this way is just by design, whether to apply the regularization is still depending on the user?  

```
...

# Initializers and regularizers for the weights and the biasses.
initializers={"w": tf.truncated_normal_initializer(stddev=1.0),
              "b": tf.truncated_normal_initializer(stddev=1.0)}
regularizers = {"w": tf.contrib.layers.l1_regularizer(scale=0.1),
                "b": tf.contrib.layers.l2_regularizer(scale=0.1)}

linear_regression_module = snt.Linear(output_size=FLAGS.output_size,
                                      initializers=initializers,
                                      regularizers=regularizers)

...

```

My second question is, suppose there are two neural networks. Each network needs to be tuned separately. I just wonder how can I retrieve the regularization loss for each neural network separately? I suspect the following way will obtain the regularization for both networks, yes?

```
# ...

# Get the regularization losses and add them together.
graph_regularizers = tf.get_collection(tf.GraphKeys.REGULARIZATION_LOSSES)
total_regularization_loss = tf.reduce_sum(graph_regularizers)

# ...

``` 
But I do need to retrieve the loss separately since I am gonna optimize the network in round-robin way.

Thanks a lot.

**Comment by mingyr** [comment]
Via experiments, seems the problems mentioned above are not that important, so just close the thread.

**Comment by malcolmreynolds** [comment]
For clarification - the first example is just demonstrating that different variables can be regularized differently. Obviously it's up to users to determine what combination of regularizers works for their situation.

For the second example, once you acquire the regularizers for the whole graph you should be able to filter based on the string names - something like:

```
graph_regularizers = tf.get_collection(tf.GraphKeys.REGULARIZATION_LOSSES)
graph_a_reg = [reg for reg in graph_regularizers if reg.name.startswith('module_a/')
graph_b_reg = [reg for reg in graph_regularizers if reg.name.startswith('module_b/')
```

**Comment by mingyr** [comment]
Thanks a lot for the always amazing answers!
